### ISSUE TITLE
"Windows fatal exception: access violation" when using registration_colored_icp without normals

### ISSUE THREAD

**Issue by Belval** [isl-org/Open3D#2421]
**Describe the bug**

When running the colored ICP algorithm with point clouds that don't have normals, there is a "Windows fatal exception: access violation" raised.

**To Reproduce**

Here is my minimal test case:

```py
import os
import open3d as o3d

import faulthandler
faulthandler.enable()

pcds = [o3d.io.read_point_cloud(os.path.join("clouds", f)) for f in os.listdir("clouds")]
pcd1 = pcds[i]
pcd1 = pcd1.select_by_index(pcd1.remove_statistical_outlier(nb_neighbors=100, std_ratio=1.0)[1])
# Removing comment on line below with fix the access violation
#pcd1.estimate_normals(search_param=o3d.geometry.KDTreeSearchParamHybrid(radius=0.1, max_nn=30))
pcd2 = pcds[i + 1]
pcd2 = pcd2.select_by_index(pcd2.remove_statistical_outlier(nb_neighbors=100, std_ratio=1.0)[1])
# Removing comment on line below with fix the access violation
#pcd2.estimate_normals(search_param=o3d.geometry.KDTreeSearchParamHybrid(radius=0.1, max_nn=30))
reg_p2p = o3d.registration.registration_colored_icp(
    pcd1, pcd2, 0.1, [[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, 1, 0], [0, 0, 0, 1]],
    o3d.registration.ICPConvergenceCriteria(max_iteration = 20000)
)
```

**Expected behavior**

No access violation, proper message explaining that normals are required for this algorithm.

**Environment (please complete the following information):**

- Operating system: Windows 10 64bits
- Python version: 3.7.9
- Open3D version: 
- Is this remote workstation?: no
- How did you install Open3D?: pip

**Additional context**

Two clouds to reproduce the issue: [pcds.zip](https://github.com/intel-isl/Open3D/files/5319985/pcds.zip)

**Comment by yxlao** [comment]
which line causes the error?

**Comment by Belval** [comment]
The last one. I take it you were not able to reproduce the issue?

**Comment by reyanshsolis** [comment]
- Colored ICP requires pointclouds to have pre-computed normals. 

The issue is related to throwing proper errors. It will be addressed with PR #3616. 
You may use the Tensor-based Colored ICP. It is more robust and these issues are taken care of, by throwing proper errors. 
http://www.open3d.org/docs/latest/tutorial/t_pipelines/t_icp_registration.html#Colored-ICP-Registration
